File: move_videoray/src/move_unfiltered.py
#!/usr/bin/env python


'''
    Nathaniel Goldfarb 
    6/29/15
    I&E scholar
    VideoRay research group
    
    Ths program moves the videoray model in rviz using 
    data from the /usble_pose node
    based on "Using urdf with robot_state_publisher" tutorial

'''

import rospy
import roslib
import math
import tf
import outlier_filter
from geometry_msgs.msg import Twist, Vector3, Pose, PoseStamped, TransformStamped
from matplotlib import matplotlib_fname
from mpl_toolkits.mplot3d import Axes3D
import sys
from matplotlib.pyplot import plot
from numpy import mean, std
import matplotlib as mpl

import numpy as np
import pandas as pd
import random
import matplotlib.pyplot as plt
plt.ion()
import matplotlib
mpl.rc("savefig", dpi=150)
import matplotlib.animation as animation
import time

#set the robot at the center


#//move the videoray using the data from the /pose_only node
def usbl_move(pos,current):

	print 'hello'
	broadcaster = tf.TransformBroadcaster()
	if outlier_filter.filter(pos.pose.position.x,pos.pose.position.y,current.position.z):
		current.position.x = pos.pose.position.x
		current.position.y = pos.pose.position.y

	
	broadcaster.sendTransform( (current.position.x,current.position.y,current.position.z), 
								(current.orientation.x,current.orientation.y,current.orientation.z,current.orientation.w),
									 rospy.Time.now(), "odom", "body" )
#move the videoray using the data from the /pose_only node	
def pose_move(pos,current):

	#pos.position.z is in kPa, has to be convereted to depth
	# P  = P0 + pgz ----> pos.position.z = P0 + pg*z_real
	z_real = (pos.position.z - 101.325)/9.81;

	#update the movement
	broadcaster = tf.TransformBroadcaster()
	current.orientation.x = pos.orientation.x
	current.orientation.y = pos.orientation.y
	current.orientation.z = pos.orientation.z
	current.orientation.w = pos.orientation.w
	current.position.z = z_real
	broadcaster.sendTransform( (current.position.x,current.position.y,current.position.z), 
								(current.orientation.x,current.orientation.y,current.orientation.z,current.orientation.w),
									 rospy.Time.now(), "odom", "body" )



if __name__ == '__main__':
	#set up the node
	rospy.init_node('move_unfiltered', anonymous=True)
	#make a broadcaster foir the tf frame
	broadcaster = tf.TransformBroadcaster()
	#make intilial values
	
	current = Pose()
	current.position.x = 0
	current.position.y = 0
	current.position.z = 0
	current.orientation.x = 0
	current.orientation.y = 0
	current.orientation.z = 0
	current.orientation.w = 0
	#send the tf frame
	broadcaster.sendTransform( (current.position.x,current.position.y,current.position.z), 
								(current.orientation.x,current.orientation.y,current.orientation.z,current.orientation.w),
									 rospy.Time.now(), "odom", "body" )

	#listen for information
	rospy.Subscriber("/usbl_pose", PoseStamped, usbl_move,current)
	rospy.Subscriber("/pose_only", Pose, pose_move, current);
	rospy.spin()

File: move_videoray/src/move_unfiltered.cpp
/*
    Nathaniel Goldfarb 
    6/25/15
    I&E scholar
    VideoRay research group
    
    Ths program moves the videoray model in rviz using 
    data from the /usble_pose node
    based on "Using urdf with robot_state_publisher" tutorial 


*/


#include <string>
#include <ros/ros.h>
#include <sensor_msgs/JointState.h>
#include <tf/transform_broadcaster.h>
#include <geometry_msgs/Pose.h>
#include <geometry_msgs/PoseStamped.h>



//global varibles
geometry_msgs::TransformStamped odom_trans;

ros::Publisher joint_pub;

//declarations
void usbl_move( const geometry_msgs::PoseStamped& pos);
void pose_move( const geometry_msgs::Pose& pos);

int main(int argc, char** argv) 
{   
    //set up 
    ros::init(argc, argv, "move_unfiltered");
    odom_trans.header.frame_id = "odom";
    odom_trans.child_frame_id = "body";
    ros::NodeHandle n;
    
    ros::Rate loop_rate(30);
    //define  the links
    ros::Subscriber usbl_sub = n.subscribe("/usbl_pose", 1, usbl_move);
    ros::Subscriber pose_sub = n.subscribe("/pose_only", 1, pose_move);
    tf::TransformBroadcaster broadcaster;
    odom_trans.header.stamp = ros::Time::now();

    broadcaster.sendTransform(odom_trans);
   

    ros::spin();


    return 0;
}
//move the videoray using the data from the /pose_only node
void pose_move( const geometry_msgs::Pose& pos)
{
    //pos.position.z is in kPa, has to be convereted to depth
    // P  = P0 + pgz ----> pos.position.z = P0 + pg*z_real
    float z_real = -1*(pos.position.z -101.325)/9.81;
   //update the movement
    tf::TransformBroadcaster broadcaster;
    odom_trans.transform.translation.z = z_real;
    odom_trans.transform.rotation.x = pos.orientation.x;
    odom_trans.transform.rotation.y = pos.orientation.y;
    odom_trans.transform.rotation.z = pos.orientation.z;
    odom_trans.transform.rotation.w = pos.orientation.w;
    //publish the movement
    joint_pub.publish(joint_state);
    broadcaster.sendTransform(odom_trans);
}

//move the videoray using the /usbl_pose node
void usbl_move( const geometry_msgs::PoseStamped& pos)
{
    //update the movement
    tf::TransformBroadcaster broadcaster;
    odom_trans.header.stamp = ros::Time::now();
    odom_trans.transform.translation.x =  pos.pose.position.x;
    odom_trans.transform.translation.y =  pos.pose.position.y;
    //publish the movement
   
    broadcaster.sendTransform(odom_trans);
    
}